ntbs-service/Services/AuthorizationService.cs

```diff
@@ -184,26 +187,33 @@ public async Task<IQueryable<Notification>> FilterNotificationsByUserAsync(Claim
 
         public async Task<IList<Alert>> FilterAlertsForUserAsync(ClaimsPrincipal user, IList<Alert> alerts)
         {
-            var userTbServiceCodes = (await _userService.GetTbServicesAsync(user)).Select(s => s.Code);
+            var userTbServiceCodes = (await _userService.GetTbServicesAsync(user)).Select(s => s.Code).ToList();
             var userType = _userService.GetUserType(user);
-            for (int i = alerts.Count - 1; i >= 0; i--)
+            alerts = alerts.Where(a => AlertIsForNotificationInOrTransferToGivenServices(a, userTbServiceCodes)).ToList();
+            if (userType == UserType.PheUser)

```

Comment:
I think the behaviour has changed here for PHE users. Previously it looks like PHE users were not filtering based on TB Service, but now I think they are. Was this intentional?

As an aside, I would have hoped that such a change would be caught by the tests. Is this code tested? It looks like we may need to add/update tests for this.

Reply:
I think the PHE users alerts were still being filtered by TBService. As the first check used to be `user is PHE & alert is transfer` this meant that all alerts that weren't transfers were still hitting the next if and getting filtered. Tests all pass.

Reply:
Oh yes, you're right! Nevermind :-)